### PR TITLE
[SPARK-23459][SQL] Improve the error message when unknown column is specified in partition columns

### PR DESCRIPTION
**Pull request by kiszk**
## What changes were proposed in this pull request?

This PR avoids to print schema internal information when unknown column is specified in partition columns. This PR prints column names in the schema with more readable format.

The following is an example.

Source code
```
test("save with an unknown partition column") {
  withTempDir { dir =>
    val path = dir.getCanonicalPath
      Seq(1L -> "a").toDF("i", "j").write
        .format("parquet")
        .partitionBy("unknownColumn")
        .save(path)
  }
```
Output without this PR
```
Partition column unknownColumn not found in schema StructType(StructField(i,LongType,false), StructField(j,StringType,true));
```

Output with this PR
```
Partition column unknownColumn not found in schema struct<i:bigint,j:string>;
```

## How was this patch tested?

Manually tested

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala [hunk 1]
```diff
@@ -486,7 +486,8 @@ object PartitioningUtils {
     val equality = columnNameEquality(caseSensitive)
     StructType(partitionColumns.map { col =>
       schema.find(f => equality(f.name, col)).getOrElse {
-        throw new AnalysisException(s"Partition column $col not found in schema $schema")
+        val schemaCatalog = schema.catalogString
+        throw new AnalysisException(s"Partition column $col not found in schema $schemaCatalog")
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala]: Quote `$col`?

Could we add a test case?

**Comment by kiszk** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala [hunk 2]
```diff
@@ -491,6 +491,22 @@ class FileSourceStrategySuite extends QueryTest with SharedSQLContext with Predi
     }
   }
 
+  test("SPARK-23459: Improve error message when specfieed unknown column in partition columns") {
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala]: move it to SaveLoadSuite?

**Comment by mgaido91** [comment]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master

**Pull request closed without merging**